EIPS/eip-4788.md

```diff
@@ -1,140 +1,109 @@
 
-Store each beacon chain state root into a contract and add a new opcode that reads this contract.
+Store each of these roots in a contract that lives in the execution state and add a new opcode that reads this contract.
 
 ## Motivation
 
-Exposing the beacon chain state root allows for proofs about the beacon state to be verified inside the EVM. This functionality supports a wide variety of use cases in smart contracts involving validator status and finality produced by the consensus layer.
-
-In particular, this functionality is required for beacon chain validator withdrawals to the EVM.
+Roots of the beacon chain blocks are crytographic accumulators that allow proofs of arbitrary consensus state. Exposing these roots inside the EVM allows for trust-minimized access to the consensus layer. This functionality supports a wide variety of use cases that improve trust assumptions of staking pools, restaking constructions, smart contract bridges, MEV mitigations and more.
 
 ## Specification
 
-| constants                 | value                                        | units
-|---                        |---                                           |---
-| `FORK_TIMESTAMP`          | TBD                                          |
-| `FORK_EPOCH`              | TBD                                          |
-| `HISTORY_STORAGE_ADDRESS` | `0xfffffffffffffffffffffffffffffffffffffffd` |
-| `OPCODE_VALUE`            | `0x48`                                       |
-| `G_beacon_state_root`     | 20                                           | gas
+| constants                   | value                                        | units
+|---                          |---                                           |---
+| `FORK_TIMESTAMP`            | TBD                                          |
+| `HISTORY_STORAGE_ADDRESS`   | `0xfffffffffffffffffffffffffffffffffffffffd` |
+| `OPCODE_VALUE`              | `0x48`                                       |
+| `G_beacon_root`             | 20                                           | gas
+| `SLOTS_PER_HISTORICAL_ROOT` | 8192                                         | slot(s)
 
 ### Background
 
-The method of injecting the beacon state root in this EIP follows the general strategy of [EIP-4399](./eip-4399.md) to make a post-merge change to the EVM integrating information from the beacon chain. This EIP along with [EIP-3675](./eip-3675.md) should be taken as relevant background to understand the particular approach of this EIP.
-
-The method for exposing the state root data via opcode is inspired by [EIP-2935](./eip-2935.md).
+The high-level idea is that each execution block contains the previous beacon block root. In the event of missed slots, the previous block root does not change
+and so we only need a constant amount of space to represent this "oracle" in each execution block. To improve the usability of this oracle, block roots are stored
```

Comment:
```suggestion
The high-level idea is that each execution block contains the parent beacon block root. Even in the event of missed slots  since the previous block root does not change,
we only need a constant amount of space to represent this "oracle" in each execution block. To improve the usability of this oracle, block roots are stored
```